pkg/network/iface/bond.go

```diff
@@ -110,12 +110,23 @@ func (b *Bond) ensureBondSlaves() error {
 	return nil
 }
 
-func (b *Bond) DeleteBond() error {
-	if _, err := netlink.LinkByName(b.Name); err != nil && !errors.As(err, &netlink.LinkNotFoundError{}) {
+func (b *Bond) remove() error {
+	slaves, err := getSlaves(b.Index)
+	if err != nil {
+		return err
+	}
+
+	if err := netlink.LinkDel(b); err != nil {
 		return err
 	}
 
-	return netlink.LinkDel(b)
+	for _, slave := range slaves {
+		if err := netlink.LinkSetUp(slave); err != nil {
```

Comment:
Question:
If some slave is `down` before removing bond, will it cause failure here?

Reply:
No, the bond could be deleted even though all slaves are down.